network/wsNetwork_test.go

```diff
@@ -1432,13 +1424,16 @@ func TestPeeringWithIdentityChallenge(t *testing.T) {
 	require.True(t, ok)
 	gossipC, err := netC.addrToGossipAddr(addrC)
 	require.NoError(t, err)
-	addrC = hostAndPort(addrC)
 
+	assert.Equal(t, 1, len(netA.GetPeers(PeersConnectedOut)))
 	// A connects to C (but uses addrB here to simulate case where B & C have the same PublicAddress)
 	netA.wg.Add(1)
 	netA.tryConnect(addrB, gossipC)
 	// let the tryConnect go forward
 	time.Sleep(250 * time.Millisecond)
+	assert.Eventually(t, func() bool {
+		return len(netA.GetPeers(PeersConnectedOut)) == 2
+	}, time.Second, 50*time.Millisecond)
```

Comment:
This time window might be a risk for the build server

Reply:
actually, good catch, I need to remove the `time.Sleep(250 * time.Millisecond)` above and leave only this retrying logic. Note, 1 sec is greater than the 250ms timeout before.